edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/authentication/AuthAgentProtocolHead.cs

```diff
@@ -95,13 +95,15 @@ public async Task CloseAsync(CancellationToken token)
         }
 
         public void Dispose()
+        {
+            DisposeAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+        }
```

Comment:
If the solution was to switch to IAsyncDisposable, and for this we only need to implement DisposeAsync, why do we need this block?

Reply:
It's a good point. It is suggested in the docs for a class to implement both, but we don't really need sync version of `Dispose`. We don't need it by two reasons: 
- aspnet DI will use IAsyncDisposable if available
- Tests must use `async using` to avoid deadlocks.
I wonder if by removing this we'll force consumers to use `await using` only - what would be a good thing.